File: backend/member-service/src/main/java/com/example/memberservice/member/dto/MemberDto.java
package com.example.memberservice.member.dto;

import com.example.memberservice.member.entity.Member;
import com.example.memberservice.member.entity.Role;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;



@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MemberDto implements Serializable {

    private Long id;
    private String userId;
    private String auth;
    private Role role;
    private String email;
    private String username;
    private String statusMessage;
    private String profileImage;
    private String wallpaperImage;

    public MemberDto(Member member) {
        setId(member.getId());
        setUserId(member.getUserId());
        setAuth(member.getAuth());
        setRole(member.getRole());
        setEmail(member.getEmail());
        setUsername(member.getUsername());
        setStatusMessage(member.getStatusMessage());
        setProfileImage(member.getProfileImage());
        setWallpaperImage(member.getWallpaperImage());
    }

    public MemberDto(GoogleAccountDto account) {
        setUserId(account.getSub());
        setEmail(account.getEmail());
        setAuth("google");
        setRole(Role.ROLE_MEMBER);
        setUsername(account.getName());
        setStatusMessage("");
        setProfileImage(account.getPicture() == null ? "" : account.getPicture());
        setWallpaperImage("");
    }

    public static MemberDto createMemberDto(Member member) {
        return MemberDto.builder()
                .id(member.getId())
                .userId(member.getUserId())
                .auth(member.getAuth())
                .role(member.getRole())
                .email(member.getEmail())
                .username(member.getUsername())
                .statusMessage(member.getStatusMessage())
                .profileImage(member.getProfileImage())
                .wallpaperImage(member.getWallpaperImage())
                .build();
    }
}

File: backend/member-service/src/main/kotlin/com/example/memberservice/member/dto/MemberDto.kt
package com.example.memberservice.member.dto

import com.example.memberservice.member.entity.Member
import com.example.memberservice.member.entity.Role
import java.io.Serializable

data class MemberDto(
    var id: Long? = null,
    var userId: String? = null,
    var auth: String? = null,
    var role: Role? = null,
    var email: String? = null,
    var username: String? = null,
    var statusMessage: String? = null,
    var profileImage: String? = null,
    var wallpaperImage: String? = null,
) : Serializable {

    constructor(member: Member) : this(
        id = member.id,
        userId = member.userId,
        auth = member.auth,
        role = member.role,
        email = member.email,
        username = member.username,
        statusMessage = member.statusMessage,
        profileImage = member.profileImage,
        wallpaperImage = member.wallpaperImage,
    )

    constructor(account: GoogleAccountDto) : this(
        userId = account.sub,
        email = account.email,
        auth = "google",
        role = Role.ROLE_MEMBER,
        username = account.name,
        statusMessage = "",
        profileImage = account.picture ?: "",
        wallpaperImage = "",
    )

    companion object {
        @JvmStatic
        fun createMemberDto(member: Member): MemberDto = MemberDto(member)
    }
}
